specs/WorkerPostMessage.md

```diff
@@ -0,0 +1,682 @@
+        CoreWebView2DedicatedWorkerCreatedEventArgs args)
+{
+    CoreWebView2DedicatedWorker dedicatedWorker = args.Worker;
+    MessageBox.Show("Dedicated worker is created" , "Dedicated Worker Message");
+    DedicatedWorker_PostMessage_SetupEventsOnDedicatedWorker(dedicatedWorker);
+    DedicatedWorker_PostMessage_ComputeWithDedicatedWorker(dedicatedWorker);
+}
+
+void DedicatedWorker_PostMessage_SetupEventsOnDedicatedWorker(
+        CoreWebView2DedicatedWorker dedicatedWorker)
+{
+    dedicatedWorker.WebMessageReceived += (sender, args) =>
+    {
+        StringBuilder messageBuilder = new StringBuilder();
+        messageBuilder.AppendLine($"Dedicated Worker: \n{args.Source} ");
+        messageBuilder.AppendLine($"\nMessage: \n{args.TryGetWebMessageAsString()} ");
+        MessageBox.Show(messageBuilder.ToString(), "Message from Dedicated Worker",
+                        MessageBoxButton.OK);
+    };
+}
+
+void DedicatedWorker_PostMessage_ComputeWithDedicatedWorker(
+        CoreWebView2DedicatedWorker dedicatedWorker)
+{
+    var dialog = new TextInputDialog(
+        title: "Post Web Message JSON",
+        description: "Enter the web message as JSON.",
+        defaultInput: "");
+    // Ex: {"command":"MUL","first":2,"second":3}
+    if (dialog.ShowDialog() == true)
+    {
+        dedicatedWorker.PostWebMessageAsJson(dialog.Input.Text);
+    }
+}
+```
+
+## Service Worker
+## Posting messages to and from a service worker
+### main.js
+```JS
+if ("serviceWorker" in navigator) {
+  navigator.serviceWorker.register("sw.js");
+}
+
+```
+### sw.js
+```JS
+'use strict';
+
+const CACHE_NAME = 'sw_post_message_cache';
+const CACHE_LIST = ['style.css'];
+
+const cacheFirst = async (request) => {
+  // First try to get the resource from the cache
+  const responseFromCache = await caches.match(request);
+  if (responseFromCache) {
+    console.log('Cache hit for request: ', request.url);
+    // Notify the app about the cache hit.
+    //! [chromeWebView]
+    self.chrome.webview.postMessage('Cache hit for resource: ' + request.url);
+    //! [chromeWebView]
+    return responseFromCache;
+  }
+
+  // Next try to get the resource from the network
+  try {
+    console.log('Cache miss for request: ', request.url);
+    const responseFromNetwork = await fetch(request);
+    const cache = await caches.open(CACHE_NAME);
+    console.log('Cache new resource: ', request.url);
+    await cache.put(request, responseFromNetwork.clone());
+    return responseFromNetwork;
+  } catch (error) {
+    return new Response('Network error happened', {
+      status: 408,
+      headers: { 'Content-Type': 'text/plain' },
+    });
+  }
+};
+
+const addToCache = async (url) => {
+  console.log('Add to cache: ', url);
+  const cache = await caches.open(CACHE_NAME);
+  cache.add(url);
+  //! [chromeWebView]
+  chrome.webview.postMessage('Added to cache: ' + url);
+  //! [chromeWebView]
+};
+
+self.addEventListener('install', (event) => {

```

Comment:
Why the sample difference between self.chrome.webview.addEventListener and self.addEventListener ?  Is this an existing asymmetry of dedicated v service workers?